### PR TITLE
Add headers from config

### PR DESCRIPTION
**Pull request by milesjos**
Hello, we are trying to add an authentication header to the connect websocket call and to do this we merged the headers being sent and the client headers passed into the constructor. Let us know if there is anything else we need to do to get this merged in. Thanks!

### CODE REVIEW

**Comment by IsNull** [comment]
Hi there, thank you for your contribution!

This is quite a good catch, and we are happy to merge this in after I checked the code.

File: src/stomp/stomp-client.ts [hunk 1]
```diff
@@ -131,11 +131,14 @@ export class StompClient {
...
 
-            const headers = new Map<string, string>();
+            let headers = new Map<string, string>();
             headers.set('accept-version', '1.2');
             headers.set('host', 'localhost');
             // headers.set('accept-version', Stomp.supportedVersions);
             // headers.set('heart-beat', [this.heartbeat.outgoing, this.heartbeat.incoming].join(','));
+            if (config && config.headers) {
+              headers = new Map([...Array.from(config.headers.entries()), ...Array.from(headers.entries())]);
+            }
```
Comment (IsNull) [inline src/stomp/stomp-client.ts]: IMHO this code would be easier to grasp and a bit shorter:
```typescript
if (config && config.headers) {
   config.headers.forEach((k, v) => headers.set(k, v));
}
```
What do you think? :)

This way, you can also leave the headers declared as `const`.

Comment (milesjos) [reply src/stomp/stomp-client.ts]: Looks good to me! Just committed the change.

**Comment by IsNull** [comment]
@milesjos I have merged and released a patch version `2.1.2` with your header fix. I have also released a new beta version which brings heart-beat support with your fix `3.0.0-beta4`.

Thanks!

**Comment by milesjos** [comment]
@IsNull Thanks!

**Pull request merged**